evento/src/views/EventPage/index.js

```diff
@@ -36,20 +54,46 @@ class EventPage extends Component {
 		});
 	}
 
+	onAttendButtonClick() {
+		const eventId = this.props.match.params.eventId;
+
+		fetch(`/events/${eventId}/attendees`, {
+			method: 'POST',
+			headers: { 'Authorization': Cookie.get('auth_token') }
+		})
+		.then(response => {
+			if (!response.ok) return Promise.reject();
+			// If successiful attending, re-fetch attendees
+			this.fetchAttendees();
+		})
+		.catch(() => this.setState({ errorMessage: "Something went wrong" }));
+	}
+
+	isUserAttending() {
+		let user = Cookie.get('user');
+		return JSON.stringify(this.state.attendees).includes(user);
```

Comment:
I agree at least in part, but since JavaScript does not provide any nice way to check equality of objects, converting to json is fairly popular. Since I already had one variable as a json it made sense. But yes, converting an array of objects to string and then match json object is a bit hacky. But still I wouldn't say it's the same as your example.

But sure, lets agree on
```js
const userCookie = Cookie.get('user');
if (!userCookie) {
    return false;
}

const user = JSON.parse(userCookie);
return this.state.attendees.some(attendee => attendee.id == user.id);
```